packages/itwinui-react/src/core/Surface/Surface.tsx

```diff
@@ -29,7 +34,59 @@ const getSurfaceElevationValue = (elevation: SurfaceProps['elevation']) => {
   }
 };
 
+// ----------------------------------------------------------------------------
+// Surface.Header component
+
+type SurfaceHeaderOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+export type SurfaceHeaderProps<T extends React.ElementType = 'div'> =
+  PolymorphicComponentProps<T, SurfaceHeaderOwnProps>;
+
+const SurfaceHeader = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', children, className, ...rest } = props;
+  return (
+    <Element
+      className={cx('iui-surface-header', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', SurfaceHeaderOwnProps>;
+
```

Comment:
What does this mean or do? To make an empty type and then export a type that is `'div'> =
  PolymorphicComponentProps<T, SurfaceHeaderOwnProps>;`

I see several instances of similar code here but I don't understand what is the significance of that.

Reply:
This is meant to make the component polymorphic, meaning that users can change the element type of their component using the `as` prop. An example of this being used can be found in the `Button -> As Link` story. More information about the `PolymorphicComponentProps` type and `PolymorphicForwardRefComponent` interface can be found [here](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/utils/props.ts#L33)

Reply:
Interesting! Still a lil confusing but I'll get it in time 😅 Thanks for explaining!